src/test/integration/checkResponseCode.py

```diff
@@ -0,0 +1,65 @@
+import json
+import pycurl
+import urllib
+import StringIO
+
+def getAccessToken(url, client_id, client_secret):
+    post_data = "client_id=" + client_id + "&client_secret=" + client_secret + "&grant_type=client_credentials"
+
+    storage = StringIO.StringIO()
+    curl    = pycurl.Curl()
+
+    # Set options
+    curl.setopt(pycurl.URL, url)                     # CURLOPT_URL in PHP
+    curl.setopt(pycurl.POST, 1)                      # CURLOPT_POST in PHP
+    curl.setopt(pycurl.POSTFIELDS, post_data)        # CURLOPT_POSTFIELDS in PHP
+    curl.setopt(pycurl.WRITEFUNCTION, storage.write) # CURLOPT_RETURNTRANSFER in PHP
+
+    # Send the request and save response
+    curl.perform()
+
+    # Close request to clear up some resources
+    curl.close()
+
+    response = storage.getvalue()
+    return json.loads(response)
+
+def getResponse(url, access_token, params):
+    query_params = urllib.urlencode(params)
+    api_call_url = url + "?" + query_params;
+    headers      = ['Authorization: Bearer '+ access_token]
+
+    storage = StringIO.StringIO()
+    curl    = pycurl.Curl()
+
+    # Set options
+    curl.setopt(pycurl.URL, api_call_url)
+    curl.setopt(pycurl.HTTPHEADER, headers) # CURLOPT_HTTPHEADER in PHP
+    curl.setopt(pycurl.WRITEFUNCTION, storage.write)
+
+    curl.perform()
+    response_code = curl.getinfo(pycurl.HTTP_CODE)
+    curl.close()
+
+    response = storage.getvalue()
+    return response_code
+
+def checkResponseCode(response_code):
+    assert response_code == 200, "Request unsuccessful"
+
+config_data_file = open('configuration.json')
+config_data      = json.load(config_data_file)
+
+base_url         = config_data["hostname"] + config_data["version"] + config_data["api"]
+access_token_url = base_url + config_data["token_endpoint"]
+
+access_token_response = getAccessToken(access_token_url, config_data["client_id"], config_data["client_secret"]);
+
+access_token  = access_token_response["access_token"]
+locations_url = base_url + config_data["locations_endpoint"]
+params        = {"q": "Oxford"}
+
+response_code = getResponse(locations_url, access_token, params)
+print "response: ", response_code
+
+checkResponseCode(response_code)
```

Comment:
Since this code will mostly be test, I would vote for leveraging a test framework for the language being used. For example, https://docs.python.org/2/library/unittest.html . What I like about using a testing framework is that we can easily define startup / setup and teardown methods that can take care of initialization of objects / deps. The teardown methods help make sure we clean up resources / objects at the end. We also get an easy report at the end knowing which tests failed or succeeded.